File: services/StatisticsProxyService/datastore/DataStore.Model/Indicator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DataStore.Model
{
    public class Indicator
    {
        public int ID { get; set; }
        public Provider Provider { get; set; }
        public string SourceID { get; set; }
        public string Name { get; set; }
        public string NameAbbr { get; set; }
        public int ThemeID { get; set; }
        public int SubThemeID { get; set; }


        public override bool Equals(object obj)
        {
            Indicator ind = obj as Indicator;
            if (ind == null) return false;

            return ID == ind.ID && Provider.Equals(Provider, ind.Provider) && String.Equals(SourceID, ind.SourceID) &&
                String.Equals(Name, ind.Name) && String.Equals(NameAbbr, ind.NameAbbr) && ThemeID == ind.ThemeID && 
                SubThemeID == ind.SubThemeID;
        }

        public override int GetHashCode()
        {
            return ID;
        }
    }
}

File: scripts/statistics/Model/Search/Indicator.js



Statistics.Model.Search.Indicator = Statistics.Class({
	
	/**
	 * @public
	 * @property {Integer}
	 * The indicator id
	 */
	id: null,

	/**
	 * @public
	 * @property {String}
	 * The indicator name
	 */	
	name: null,

	/**
	 * @public
	 * @property {String}
	 * The indicator abbreviated name
	 */	
	nameAbbr: null,

	/**
	 * @public
	 * @property {Statistics.Model.Search.Provider}
	 * The indicator provider
	 */		
	provider: null,

	/**
	 * @public
	 * @property {integer}
	 * The indicator's theme code
	 */	
	themeid: null,

	/**
	 * @public
	 * @property {integer}
	 * The indicator's sub theme code
	 */		
	subthemeid: null,
	
	/**
	 * @constructor
	 * @param {integer} id
	 * @param {String} name
	 * @param {String} nameAbbr
	 * @param {String} provider
	 * @param {Integer} themeid
	 * @param {Integer} subthemeid
	 */
	_init: function(id, name, nameAbbr, provider, themeid, subthemeid) {
		this.id = id;
		this.name = name;
		this.nameAbbr = nameAbbr;
		this.provider = provider;
		this.themeid = themeid;
		this.subthemeid = subthemeid;
	}
	
});


Statistics.Model.Search.Indicator.FromObject = function(obj) {
	
	var provider = Statistics.Model.Search.Provider.FromObject(obj.Provider);
	
	return new Statistics.Model.Search.Indicator(
		obj.ID,
		obj.Name, 
		obj.NameAbbr,
		provider,
		obj.ThemeID,
		obj,SubThemeID 
	);
}
